server/server.ts

```diff
@@ -1,78 +1,105 @@
 import "dotenv/config";
 import { backendApiProxy } from "./backendApiProxy";
 import { backendApiProxyMock } from "./backendApiProxyMock";
 
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = dirname(__filename);
-
 const basePath = "/min-ia";
-const buildPath =
-  process.env.NODE_ENV === "development"
-    ? path.resolve(__dirname, "../client/out")
-    : path.resolve(__dirname, "../../client/out");
 console.log("NODE_ENV", process.env.NODE_ENV);
-console.log("buildPath", buildPath);
 
 const server = express();
 
 const envProperties = {
-  BACKEND_API_BASE_URL:
-    process.env.BACKEND_API_BASE_URL || "http://localhost:8080",
+  APP_INGRESS: process.env.APP_INGRESS || "http://localhost:3000/min-ia",
   PORT: process.env.PORT || 3010,
 };
 
+const getLoginTilOauth2 = (redirectUrl: string): string => {
+  const referrerUrl = `${envProperties.APP_INGRESS}/success?redirect=${redirectUrl}`;
+  const loginTilOAuth2 = basePath + `/oauth2/login?redirect=${referrerUrl}`;
+  return loginTilOAuth2;
+};
+
 const startServer = async () => {
   server.use(cookieParser());
   console.log("Starting server: server.js");
 
-  if (process.env.NODE_ENV === "not-local") {
+  if (process.env.NODE_ENV === "production") {
     await Promise.all([initIdporten(), initTokenX()]);
   }
 
-  server.use(basePath + "/", express.static(buildPath));
-  server.use("/assets", express.static(`${buildPath}/assets`));
-
-  if (process.env.NODE_ENV === "not-local") {
+  console.log(`NODE_ENV er '${process.env.NODE_ENV}'`);
+  if (process.env.NODE_ENV === "production") {
+    console.log("Starter backendProxy");
     server.use(backendApiProxy);
   } else {
+    console.log("Starter backendProxyMock");
     backendApiProxyMock(server);
   }
 
   server.get(`${basePath}/redirect-til-login`, (request, response) => {
-    const referrerUrl = `${process.env.APP_INGRESS}/success?redirect=${request.query.redirect}`;
-    response.redirect(basePath + `/oauth2/login?redirect=${referrerUrl}`);
+    let redirect: string = request.query.redirect
+      ? (request.query.redirect as string)
+      : envProperties.APP_INGRESS;
+
+    if (!redirect.startsWith(envProperties.APP_INGRESS)) {
+      console.log(
+        "[WARN] redirect starter ikke med APP_INGRESS, oppdaterer til ",
+        envProperties.APP_INGRESS
+      );
+      redirect = envProperties.APP_INGRESS;
+    }
+
+    console.log("Kommer inn til /redirect-til-login");
+    const loginTilOauth2 = getLoginTilOauth2(redirect);
+    console.log("[DEBUG] redirect til: ", loginTilOauth2);
+    response.redirect(loginTilOauth2);
   });
 
   server.get(`${basePath}/success`, (request, response) => {
+    /*
+     Bruker kommer til /success etter innlogging flow er fullført
+     I følge dokumentasjon skal Auth header blitt satt med en gyldig token
+     -> hva med Cookie?
+     -> skal vi sjekke denne Auth header her eller la backend håndtere det?
+     -> hvis ingen cookie eller auth header ... hva gjør vi ?
+     */
```

Comment:
Det bør egentlig ikke være nødvendig å sjekke for cookies, men det skader jo ikke. Det skal i teorien holde å sjekke for `Authorization` headeren som skal alltid være med så lenge brukeren har en sesjon. Hvis headeren ikke er med så kan man bare redirecte til login igjen, som dere allerede gjør ut fra det jeg kan se, men det er litt opp til dere å bestemme hva som skal skje her.

Reply:
Takk for dine kommentarer Trong. Da sjekker vi bare at Auth header er tilstedet og sender brukeren til login igjen ved behov. 
Validering av selve token gjøres på backend